Classes/General.cs

```diff
@@ -89,7 +90,37 @@ public static class General {
                             try {
                                 if (aborting) return;
                                 response = (HttpWebResponse)request.EndGetResponse(requestResultParam);
-                                responseStream = new ThrottledStream(response.GetResponseStream(), Settings.MaximumBytesPerSecond ?? ThrottledStream.Infinite);
+                                if (response.ContentType == ContentType.TextHtml) {
+                                    MemoryStream memoryStream = new MemoryStream();
+                                    CopyStream(response.GetResponseStream(), memoryStream);
+                                    memoryStream.Position = 0;
+                                    byte[] redirectPageBytes = memoryStream.ToArray();
+                                    Encoding pageEncoding = DetectHTMLEncoding(redirectPageBytes, response.ContentType);
+                                    metaRedirectHtml = pageEncoding.GetString(redirectPageBytes);
+                                    memoryStream.Position = 0;
+                                    responseStream = memoryStream;
+                                    if (IsMetaRefresh(metaRedirectHtml)) {
+                                        HttpWebRequest redirectionRequest = null;
+                                        redirectionRequest = (HttpWebRequest)WebRequest.Create(GetRefreshUrl(metaRedirectHtml));
+                                        if (connectionGroupName != null) {
+                                            redirectionRequest.ConnectionGroupName = connectionGroupName;
+                                        }
+                                        redirectionRequest.UserAgent = (Settings.UseCustomUserAgent == true) ? Settings.CustomUserAgent : ("Chan Thread Watch " + Version);
+                                        redirectionRequest.Referer = string.Empty; // 4Chan actively blocks some Referers such as Archived.Moe
```

Comment:
Can we simply not set this header instead of having it be the empty string?

Reply:
Do any of the boards need this? I noticed it's stored in the ImageInfo and the ThumbnailInfo when we parse the HTML.